src/axom/core/ArrayBase.hpp

```diff
@@ -1143,7 +1143,21 @@ struct DeviceStagingBuffer<T, OperationSpace::Unified_Device>
 template <typename T, OperationSpace SPACE>
 struct ArrayOpsBase
 {
-  #if defined(__CUDACC__)
+  #if !defined(AXOM_GPUCC)
+  // To avoid an ODR issue, we error out here to ensure that axom::Array
+  // device-aware operations are not instantiated within a host-compiler.
+  // See:
+  // - https://github.com/LLNL/axom/issues/1182
+  // - https://github.com/LLNL/axom/issues/1440
+  static_assert(SPACE != OperationSpace::Device,
+                "Cannot instantiate Array operations -- file is being compiled "
```

Comment:
Would it be clearer to say "device Array operations"?

Reply:
Or "device specialization of Array operations"?

Reply:
I tried to clarify the diagnostic along those lines.

Reply:
Looks good, thanks!